### PR TITLE
fix: minor basic stats quality fixes

### PR DESCRIPTION
**Pull request by ethanglaser**
## Description

A few small corrections

---

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [ ] I have run it locally and tested the changes extensively.
- [ ] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

### CODE REVIEW

File: onedal/basic_statistics/basic_statistics.py [hunk 1]
```diff
@@ -157,7 +157,7 @@ def fit(self, data, sample_weight=None, queue=None):
         data_table, weights_table = to_table(data, sample_weight, queue=queue)
 
         dtype = data_table.dtype
-        raw_result = raw_result = self._compute_raw(
+        raw_result = self._compute_raw(
```
Comment (icfaust) [inline onedal/basic_statistics/basic_statistics.py]: yikes

**Review by icfaust — COMMENTED** [review]
requires a 'black' fix. otherwise good to go

File: examples/sklearnex/basic_statistics_spmd.py [hunk 2]
```diff
@@ -48,7 +48,7 @@ def generate_data(par, size, seed=777):
 
 params_spmd = {"ns": 19, "nf": 31}
 
-data, weights = generate_data(params_spmd, size)
+data, weights = generate_data(params_spmd, rank)
```
Comment (david-cortes-intel) [inline examples/sklearnex/basic_statistics_spmd.py]: Shouldn't this be made to generate different data for each rank?

Comment (ethanglaser) [reply examples/sklearnex/basic_statistics_spmd.py]: Yes - that was the original mistake here. `size` is the same for every rank, so the same data is generated. `rank` is different on every rank, so different data is generated.

Comment (david-cortes-intel) [reply examples/sklearnex/basic_statistics_spmd.py]: But it's still being generated in a loop where each rank contains the data from the previous one.

Comment (ethanglaser) [reply examples/sklearnex/basic_statistics_spmd.py]: Although maybe that would be reflected in the `seed` parameter and this should be tweaked further. The data generation function here is pretty wonky. I'll take a closer look tomorrow.

Comment (ethanglaser) [inline examples/sklearnex/basic_statistics_spmd.py]:
```suggestion
data, weights = generate_data(params_spmd, size, seed=rank)
```

**Review by Alexsandruss — APPROVED** [review]
